src/Rings/mpoly-graded.jl

```diff
@@ -1399,27 +1399,27 @@ mutable struct HilbertData
   end
 end
 
-function hilbert_series(H::HilbertData, i::Int= 1)
+function hilbert_series(H::HilbertData)
   Zt, t = ZZ["t"]
```

Comment:
General comment: in "library code" one should never create global rings (cached = true), they should always be created with `cache = false`, ie. `Zt, t = polynomial_ring(ZZ, :t, cached = false)` 
Same is true for the Qt in the other function.
Try to do
```
Hecke.print_cache()
```
to get an idea how many of this rings/ fields are stored...
If you really want a global univariate polyring over ZZ or QQ, try `Hecke.Global.Zx` or `Hecke.Globals.Qx`

Reply:
Indeed. Actually, shouldn't the polynomial ring be an (optional) argument?

Reply:
An idea we considered for CoCoA was to store a suitable "Hilbert series ring" as a data-member of each graded polynomial ring.  This avoids asking the caller to specify the ring for the Hilbert series, and also ensures that Hilbert series of quotients of the same graded ring belong to the same Hilbert series ring.  This idea has **not yet** been implemented in CoCoA... so maybe there are unforeseen hitches.